W-12514697-amazon-dynamodb-1.5-kt (#2232)
mulesoft/docs-connectors: amazon-dynamodb/1.5/modules/ROOT/pages/amazon-dynamodb-connector-reference.adoc

| Target Variable a| String |  The name of a variable to store the operation's output. |  |
| Target Value a| String |  An expression to evaluate against the operation's output and store the expression outcome in the target variable |  `#[payload]` |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |<<ListTablesResult>>
| Attributes Type a| <<RequestIDAttribute>>
|===

=== For Configurations
* <<config>>

==== Throws
* DYNAMODB:ACCESS_DENIED_EXCEPTION
* DYNAMODB:CONDITIONAL_CHECK_FAILED_EXCEPTION
* DYNAMODB:CONNECTIVITY
* DYNAMODB:INCOMPLETE_SIGNATURE_EXCEPTION
* DYNAMODB:ITEM_COLLECTION_SIZE_LIMIT_EXCEEDED_EXCEPTION
* DYNAMODB:LIMIT_EXCEEDED_EXCEPTION
* DYNAMODB:MISSING_AUTHENTICATION_TOKEN_EXCEPTION
* DYNAMODB:PROVISIONED_THROUGHPUT_EXCEEDED_EXCEPTION
* DYNAMODB:RESOURCE_IN_USE_EXCEPTION
* DYNAMODB:RESOURCE_NOT_FOUND_EXCEPTION
* DYNAMODB:RETRY_EXHAUSTED
* DYNAMODB:THROTTLING_EXCEPTION
* DYNAMODB:UNKNOWN
* DYNAMODB:UNRECOGNIZED_CLIENT_EXCEPTION
* DYNAMODB:VALIDATION_EXCEPTION

[[paginatedQuery]]
=== Paginated Query
`<dynamodb:paginated-query>`

> The *Paginated query* operation finds items based on primary key values. You can query any table or secondary index that has a composite primary key, a partition key, and a sort key. The operation returns pages with a limit given in the *Limit* field or the maximum amount of 1MB data. This behavior differetiantes with the existing *Query* operation that truncates the response ignoring the DynamoDB value of the *Last Evaluated Key* field, which indicates that the query has more matching items. 

Reviewer: ```suggestion
The *Paginated query* operation finds items based on primary key values. You can query any table or secondary index that has a composite primary key, a partition key, and a sort key. The operation returns pages with a limit specified in the *Limit* field or the maximum amount of 1MB data. This behavior differentiates from the existing *Query* operation that truncates the response ignoring the DynamoDB value of the *Last Evaluated Key* field, which indicates that the query has more matching items. 
```

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x

> | Key Condition Expression a| String |  The condition that specifies the key values for items to be retrieved by the Query action. |  |

Reviewer: ```suggestion
| Key Condition Expression a| String |  The condition that specifies the key values for items to retrieve by the *Query* operation. |  |
```

> | Key Conditions a| Object |  This is a legacy parameter. Use KeyConditionExpression instead. |  |

Reviewer: ```suggestion
| Key Conditions a| Object |  This is a legacy parameter. Use *Key Condition Expression* instead. |  |
```

| Scan Index Forward a| Boolean |  Specifies the order for index traversal: If true (default), the traversal is performed in ascending order; if false, the traversal is performed in descending order. |  false |
| Select a| Enumeration, one of:

** ALL_ATTRIBUTES
** ALL_PROJECTED_ATTRIBUTES
** SPECIFIC_ATTRIBUTES

> ** COUNT |  The attributes to be returned in the result. |  |

Reviewer: ```suggestion
** COUNT |  The attributes to return in the result. |  |
```

| Attribute Names a| Object | One or more substitution tokens for attribute names in an expression. |  |
| Attribute Values a| Object | One or more values to substitute in an expression. |  |
| Table Name a| String | Name of the table containing the requested item. |  | x

> | Attributes To Gets a| Array of String | This is a legacy parameter. Use the Projection Expression field instead. |  |

Reviewer: ```suggestion
| Attributes To Gets a| Array of String | This is a legacy parameter. Use the *Projection Expression* field instead. |  |
```

| Exclusive Start Key a| Object | The primary key of the first item that this operation evaluates. |  |
| Filter Expression a| String | A string that contains conditions that DynamoDB applies after the operation, but before the data is returned to you. Items that do not satisfy the criteria are not returned. |  |
| Index Name a| String | The name of a secondary index to scan. This index can be any local secondary index or global secondary index. |  |
| Limit a| Number | The maximum number of items to evaluate (not necessarily the number of matching items).  |  |
| Projection Expression a| String | A string that identifies one or more attributes to retrieve from the table. |  |
| Condition Filter a| Object | The condition that evaluates the query results after the items are read and returns only the desired values. |  |
| Return Consumed Capacity a| Enumeration, one of:

** INDEXES
** TOTAL
** NONE | The level of detail of the provisioned throughput consumption returned in the response. |  |

> | Target Variable a| String |  The name of a variable to store the operation's output. |  |

Reviewer: ```suggestion
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
```

> | Target Value a| String |  An expression to evaluate against the operation's output and store the expression outcome in the target variable |  `#[payload]` |

Reviewer: ```suggestion
| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  `#[payload]` |
```

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
|===

==== Output
[cols=".^50%,.^50%"]
|======================

> | *Type* a| Array of Message of [Array of Object] payload and [<<RequestIDAttribute>>] attributes

Reviewer: ```suggestion
| *Type* a| Array of Message of Array of Object payload and <<RequestIDAttribute>> attributes
```
